core/src/main/java/org/apache/calcite/rel/metadata/RelMdDistinctRowCount.java

```diff
@@ -79,6 +82,37 @@ public Double getDistinctRowCount(RelNode rel, RelMetadataQuery mq,
     return null;
   }
 
+  public Double getDistinctRowCount(TableScan rel, RelMetadataQuery mq,
+      ImmutableBitSet groupKey, RexNode predicate) {
+    final double selectivity = predicate == null ? 1D : mq.getSelectivity(rel, predicate);
+    final boolean uniq = RelMdUtil.areColumnsDefinitelyUnique(mq, rel, groupKey);
+    if (uniq) {
+      return NumberUtil.multiply(mq.getRowCount(rel), selectivity);
```

Comment:
If the column values are unique, shouldn't we return 1 * selectivity?

Reply:
No. If the column values are unique, it does not mean only one row.